featuremanagement/aio/_featuremanager.py

```diff
@@ -108,26 +112,30 @@ def _assign_variant(self, feature_flag, targeting_context):
         :param TargetingContext targeting_context: Targeting context.
         :return: Variant name.
         """
+        evaluation_event = EvaluationEvent(feature_flag=feature_flag)
```

Comment:
I think there should be only one EvaluationEvent instance during one feature evaluation which is the instance you created in the beginning of `_check_feature` method.

And all methods should modify that instance in place instead of returning a new instance or the EvaluationEvent instance passed in as a parameter.

Reply:
This should be updated everywhere